python-package/lightgbm/sklearn.py

```diff
@@ -2,6 +2,7 @@
 """Scikit-learn wrapper interface for LightGBM."""
 import copy
 from inspect import signature
+from joblib import cpu_count
```

Comment:
We don't have `joblib` in our Python-package requirements. It's available just because it's a dependency of scikit-learn. I'm afraid it can be changed in the future.
https://github.com/microsoft/LightGBM/blob/417c732cc05978276d533c43070d24dfd6c892b7/python-package/setup.py#L351-L356

Moreover, we try to keep other LightGBM interfaces working even if there is no scikit-learn installed in environment.
https://github.com/microsoft/LightGBM/blob/417c732cc05978276d533c43070d24dfd6c892b7/python-package/lightgbm/sklearn.py#L471-L473

I really like how `joblib` determines number of CPUs in their complex way. I guess we can have something the following in [our compatibility module](https://github.com/microsoft/LightGBM/blob/master/python-package/lightgbm/compat.py)

```python
try:
    from joblib import cpu_count
except ImportError:
    try:
        from psutil import cpu_count
    except ImportError:
        from multiprocessing import cpu_count
```

WDYT?